enterprise/views.py

```diff
@@ -1124,16 +1342,46 @@ def get_program_details(self, request, program_uuid, enterprise_customer):
         """
         try:
             course_catalog_api_client = CourseCatalogApiServiceClient(enterprise_customer.site)
-        except ImproperlyConfigured:
-            raise Http404
+        except ImproperlyConfigured as error:
+            error_code = 'ENTPEV002'
+            LOGGER.error(
+                'Error: {error}. Returned error code {error_code} to user'.format(
+                    error=error,
+                    error_code=error_code,
+                )
+            )
+            messages.add_generic_error_message_with_code(request, error_code)
+            return ({}, error_code)
 
         program_details = course_catalog_api_client.get_program_by_uuid(program_uuid)
         if program_details is None:
-            raise Http404
+            error_code = 'ENTPEV003'
+            LOGGER.error(
+                'A user of enterprise customer {enterprise_customer} encountered an error. '
+                'program_details is None for program_uuid {program_uuid}. '
+                'Returned error code {error_code} to user'.format(
+                    enterprise_customer=enterprise_customer.uuid,
+                    program_uuid=program_uuid,
+                    error_code=error_code,
+                )
+            )
+            messages.add_generic_error_message_with_code(request, error_code)
+            return ({}, error_code)
 
         program_type = course_catalog_api_client.get_program_type_by_slug(slugify(program_details['type']))
         if program_type is None:
-            raise Http404
+            error_code = 'ENTPEV004'
+            LOGGER.error(
+                'A user of enterprise customer {enterprise_customer} encountered an error. '
+                'program_type is None for program_details of program_uuid {program_uuid}. '
+                'Returned error code {error_code} to user'.format(
+                    enterprise_customer=enterprise_customer.uuid,
+                    program_uuid=program_uuid,
+                    error_code=error_code,
+                )
+            )
+            messages.add_generic_error_message_with_code(request, error_code)
+            return ({}, error_code)
```

Comment:
because there are a few embedded methods here that normally return dictionaries, we want to pass up a blank dictionary and the error_code so we can bubble up the errors and catch them in the respective GET/POST